### PR TITLE
Use foreman remote execution to run salt state.highstate

### PR DESCRIPTION
**Pull request by sbernhard**

### CODE REVIEW

File: app/views/foreman_salt/job_templates/salt_run_apply_highstate_-_ssh_default.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  required: true
+  input_type: user
+  description: Enables test run
+  options: "no\r\nyes"
+  advanced: true
+kind: job_template
+%>
+
+salt-call state.highstate \
+<%= ' test=true' if input('test-run') == 'yes' -%>
```
Comment (adamruzicka) [inline app/views/foreman_salt/job_templates/salt_run_apply_highstate_-_ssh_default.erb]: This template is a specialization of the `Salt Run Function` template and could be made a bit DRYer using foreign input sets (like in package actions in katello[1])

[1] - https://github.com/Katello/katello/blob/master/app/views/foreman/job_templates/remove_package.erb

File: lib/foreman_salt/engine.rb [hunk 2]
```diff
@@ -63,4 +63,9 @@ class Engine < ::Rails::Engine
       require 'foreman_salt/extensions'
     end
   end
+
+  # check whether foreman_remote_execution to integrate is available in the system
+  def self.with_remote_execution?
+    RemoteExecutionFeature rescue false
```
Comment (adamruzicka) [inline lib/foreman_salt/engine.rb]: Wouldn't `defined? RemoteExecutionFeature` do the same thing?

File: lib/foreman_salt/plugin.rb [hunk 3]
```diff
@@ -144,5 +144,14 @@
   parameter_filter Host::Managed, :salt_proxy_id, :salt_proxy_name,
     :salt_environment_id, :salt_environment_name, :salt_modules => [],
     :salt_module_ids => []
+
+  if ForemanSalt.with_remote_execution? && Gem::Version.new(ForemanRemoteExecution::VERSION) >= Gem::Version.new('1.2.3')
```
Comment (adamruzicka) [inline lib/foreman_salt/plugin.rb]: Why specifically `REX >= 1.2.3`?

File: lib/foreman_salt/engine.rb (hunk 2 above)
Comment (sbernhard) [reply lib/foreman_salt/engine.rb]: adapted from foreman_openscap plugin and similar to foreman_chef.

File: lib/foreman_salt/plugin.rb (hunk 3 above)
Comment (sbernhard) [reply lib/foreman_salt/plugin.rb]: adapted from foreman_openscap plugin and similar to foreman_chef.

File: app/views/foreman_salt/job_templates/salt_run_apply_highstate_-_ssh_default.erb (hunk 1 above)
Comment (sbernhard) [reply app/views/foreman_salt/job_templates/salt_run_apply_highstate_-_ssh_default.erb]: good idea. will have a look on it. thanks.

**Comment by sbernhard** [comment]
Added a setting to enable / disable the "Run Salt" button on the host details page. The default value of this setting is true in case RemoteExecution plugin is available.

**Comment by sbernhard** [comment]
[test foreman_salt]

**Comment by brimioulle** [comment]
Successfully tested features:
- new salt report logo
- running salt via "Run Salt-button"
- running salt high-state via remote execution (ssh default)
- running salt function (cmd.run 'df -h', test.ping) via remote execution

**Pull request merged**